insight/tasks/weekly_trend_analysis.py

```
@@ -0,0 +1,104 @@
+        except Exception as e:
+            logger.warning("Failed to fetch post detail (id=%s, attempt %d): %s", post_id, attempt + 1, e)
+            if attempt < max_retries - 1:
+                await asyncio.sleep(2**attempt)
+    return None
+
+
+async def retry_analyze_trending_posts(payload, api_key, max_retries=3):
+    for attempt in range(max_retries):
+        try:
+            return analyze_trending_posts(payload, api_key)
+        except Exception as e:
```

Ruff (0.11.9)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)